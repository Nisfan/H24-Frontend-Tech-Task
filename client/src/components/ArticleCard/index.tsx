import React from "react";

import { Article } from "../../types";
import { currencyFormatter } from "../../utils/formatter";

type ArticleCardProps = {
  article: Article;
};

const ArticleCard = ({ article }: ArticleCardProps) => {
  return (
    <li className="article">
      <img src={article.images[0].path} alt="article" />
      <h2 className="name">{article.name}</h2>
      <p>{currencyFormatter.format(article.prices.regular.value / 100)}</p>
      <button>Add to cart</button>
    </li>
  );
};

export default ArticleCard;
