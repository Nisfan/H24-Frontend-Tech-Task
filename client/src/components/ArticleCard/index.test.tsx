import React from "react";
import { create } from "react-test-renderer";

import "@testing-library/jest-dom";
import ArticleCard from ".";
import { Article } from "../../types";

const article: Article = {
  name: "name",
  variantName: "variant name",
  prices: { currency: "EUR", regular: { value: 100 } },
  images: [{ path: "path" }],
};

test("render article card", async () => {
  const tree = create(<ArticleCard article={article} />).toJSON();

  expect(tree).toMatchSnapshot();
});
