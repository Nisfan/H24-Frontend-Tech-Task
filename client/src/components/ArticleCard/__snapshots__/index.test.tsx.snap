// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render article card 1`] = `
<li
  className="article"
>
  <img
    alt="article"
    src="path"
  />
  <h2
    className="name"
  >
    name
  </h2>
  <p>
    1,00 €
  </p>
  <button>
    Add to cart
  </button>
</li>
`;
